avr-hal-generic/src/i2c.rs

```diff
@@ -499,5 +499,82 @@ macro_rules! impl_twi_i2c {
                 Ok(())
             }
         }
+
+        $(#[$i2c_attr])*
+        pub struct Slave<CLOCK: $crate::clock::Clock, M> {
```

Comment:
Need to call this something besides `Slave`. You had suggested `$I2cSlave` but that caused errors in the macro expansion or subsequent build.